tests/ignite/metrics/test_rouge.py

```diff
@@ -0,0 +1,195 @@
+        rouge = Rouge(n="m")
+
+    with pytest.raises(ValueError):
+        rouge = Rouge(n=-1)
+
+
+def test_compute():
+    rouge = Rouge()
+
+    y_pred = "the cat was found under the bed"
+    y = "the cat was under the bed"
+    rouge.update([y_pred.split(), [y.split()]])
+    assert isinstance(rouge.compute(), torch.Tensor)
+    assert rouge.compute() == 0.8571428656578064
+
+    y_pred = "the tiny little cat was found under the big funny bed"
+    y = "the cat was under the bed"
+    rouge.update([y_pred.split(), [y.split()]])
+    assert isinstance(rouge.compute(), torch.Tensor)
+    assert rouge.compute() == 0.701298713684082
+
+    rouge = Rouge(n="l")
+
+    y_pred = "the cat was found under the bed"
+    y = "the cat was not under the bed"
+    rouge.update([y_pred.split(), [y.split()]])
+
+    assert isinstance(rouge.compute(), torch.Tensor)
+    assert rouge.compute() == 0.8571428656578064
+
+    y_pred = "the cat was found under the big funny bed"
+    y = "the tiny little cat was under the bed"
+    rouge.update([y_pred.split(), [y.split()]])
+
+    assert isinstance(rouge.compute(), torch.Tensor)
+    assert rouge.compute() == 0.761904776096344
+
+
+def _test_distrib_integration(device):
+    import ignite.distributed as idist
+    from ignite.engine import Engine
+    from ignite.metrics import Rouge
+
+    n_iters = 2
+
+    y_true = ["Hi", "Hello"]
+    y_preds = [["Hi there!", "Hi , How are you?"], ["Hello there", "Hello , How are you?"]]
+
+    def update(engine, i):
+        return (y_true[i].split(), [s.split() for s in y_preds[i]])
+
+    def _test_n(metric_device):
+        engine = Engine(update)
+        m = Rouge(device=metric_device)
+        m.attach(engine, "rouge")
+
+        data = list(range(n_iters))
+        engine.run(data=data, max_epochs=1)
+
+        assert "rouge" in engine.state.metrics
+
+    def _test_l(metric_device):
+        engine = Engine(update)
+        m = Rouge(n="l", device=metric_device)
+        m.attach(engine, "rouge")
+
+        data = list(range(n_iters))
+        engine.run(data=data, max_epochs=1)
+
+        assert "rouge" in engine.state.metrics
+
+    _test_n("cpu")
+    _test_l("cpu")
+    if device.type != "xla":
+        _test_n(idist.device())
+        _test_l(idist.device())
+
+
+def _test_distrib_accumulator_device(device):
+
+    metric_devices = [torch.device("cpu")]
+    if torch.device(device).type != "xla":
+        metric_devices.append(idist.device())
+
+    for metric_device in metric_devices:
+        rouge = Rouge(device=metric_device)
+        dev = rouge._device
+        assert dev == metric_device, f"{dev} vs {metric_device}"
+
+        y_pred = "the tiny little cat was found under the big funny bed"
+        y = "the cat was under the bed"
+        rouge.update([y_pred.split(), [y.split()]])
+        dev = rouge._rougetotal.device
+        assert dev == metric_device, f"{dev} vs {metric_device}"
+
+
+# def test_accumulator_detached():
```

Comment:
@gucifer please remove useless code.

Reply:
This is another question I wanted to ask. What does this function exactly do in other metrics and is it useful for rouge score?

Reply:
This test checks that your tensors inside the metric do not need gradients. It would be relevant to have such validation.

Reply:
Let me add that function to the tests then.